### DISCUSSION TITLE
lxqt without systemd

### DISCUSSION THREAD

**Discussion by edefranco** [lxqt/lxqt#1997] [General]
I make a small system with debootstrap an I would like to install lxqt, but lxqt-core require some dependency (lxqt-panel -> lxqt-policykit -> libpolkit-qt5-1-1) that require systemd.

I don't like systemd and I would like sysvinit: why I am obligate to install systemd ?

Is possible to install lxqt without systemd ?

Emilio

**Comment by stefonarch** [comment]
Only thing I know: https://artixlinux.org/ has LXQt so it should be possible.

**Reply by tsujan** [reply]
systemd forced its way into Linux, torturing many users with its nasty issues (although it has been stable for a while). So, in most distros — especially in those based on Debian — it's hard to avoid systemd. In Arch, it should be possible because an Arch user can customize his system from bottom to top but I don't think it's a piece of cake either.

However, some distros make systemd optional or avoid it by default (like Gentoo and Artix Linux) and still, they have LXQt. You might want to try or consult them.

**Comment by Chiitoo** [comment]
It is not only possible, but quite easy to do as long as you don't need `backlight` control, which `liblxqt` uses `polkit-qt` for.  This can be disabled at build time.

Of course if you do want `backlight` control, then things will not be as easy (thought you could at least replace systemd with elogind, if it makes any difference).

**Comment by edefranco** [comment]
Ok Chiitoo, but: how to do ? I don't need backlight control but systemd is requeired from dependency of lxqt core (as mentioned, lxqt core request lxqt panel that request lxqt-policykit thet request libpolkit-qt5-1-1 that request systemd).
I have not mode to not install lxqt without systemd ! 

I use debian busty as referement whit deboostrap because for me is the better system to make small personalized distro.

Emilio

**Reply by tsujan** [reply]
LXQt doesn't have a component called "core". That's just a package in your distro. When you don't want systemd, you can't use the packages provided by Debian.

**Reply by edefranco** [reply]
Ok, thanks for info ... the problem is the .deb package that is make with the dependency ... I need to make a project from source ... not with deboostrap the is a simple way to recreate a debian mini distro but with all "restriction" of it ...

Emilio

**Comment by darkshram** [comment]
@edefranco If it helps, I maintain a LXQt set of RPM packages for a **non-SystemD** fork of Fedora, so I can give you some advice. My Linux distribution (ALDOS) uses elogind + eudev instead of SystemD. I have been maintaining LXQt packages for ALDOS since LXQt came to light, and @tsujan can confirm I'm usually relatively active every new LXQt release (because we use QT 5.12 LTS). We make a lot of testing and validations before releasing packages in our software repositories.

So, the short answer is: **YES**, it is possible to build and run LXQt without SystemD. Requires a lot of work? Yes.

To build LXQt packages without SystemD you will need to install runtime and development packages for:

* eudev >= 3.2.6
* elogind >= 2.39.5
* sysvinit >= 2.98
* Qt >= 5.12.10

Since you are doing it from a Debian derivative, I strongly suggest to use as base the packages from **Devuan** (a Debian fork without SystemD). You will have half the work done if you use Devuan as base and avoid ALL packages from Debian.
